tests/entities/repos/test_submission_repo.py

```diff
@@ -216,25 +226,56 @@ async def test_get_filing_period(self, query_session: AsyncSession):
         assert res.filing_type == FilingType.ANNUAL
 
     async def test_add_filing(self, transaction_session: AsyncSession):
-        res = await repo.create_new_filing(transaction_session, lei="12345ABCDE", filing_period="2024")
+        user_action_create = await repo.add_user_action(
+            transaction_session,
+            user_id="123456-7890-ABCDEF-GHIJ",
+            user_name="test creator",
+            user_email="test@local.host",
+            action_type=UserActionType.CREATE,
+        )
+        res = await repo.create_new_filing(
+            transaction_session, lei="12345ABCDE", filing_period="2024", creator_id=user_action_create.id
+        )
         assert res.id == 4
         assert res.filing_period == "2024"
         assert res.lei == "12345ABCDE"
         assert res.institution_snapshot_id is None
+        assert res.creator.id == user_action_create.id
+        assert res.creator.user_id == "123456-7890-ABCDEF-GHIJ"
+        assert res.creator.user_name == "test creator"
+        assert res.creator.user_email == "test@local.host"
+        assert res.creator.action_type == UserActionType.CREATE
 
     async def test_modify_filing(self, transaction_session: AsyncSession):
-        mod_filing = FilingDTO(
+        user_action_create = UserActionDAO(
+            id=4,
+            user_id="123456-7890-ABCDEF-GHIJ",
+            user_name="Test Creator",
+            user_email="test@local.host",
+            action_type=UserActionType.CREATE,
+            timestamp=datetime.datetime.now(),
+        )
+        """mod_filing = FilingDTO(
+            id=3, lei="ZYXWVUTSRQP", institution_snapshot_id="Snapshot-2", filing_period="2024", tasks=[], creator_id=4
+        )"""
+
+        mod_filing = FilingDAO(
             id=3,
             lei="ZYXWVUTSRQP",
             institution_snapshot_id="Snapshot-2",
             filing_period="2024",
             tasks=[],
+            creator=user_action_create,
         )
+
         res = await repo.upsert_filing(transaction_session, mod_filing)
         assert res.id == 3
         assert res.filing_period == "2024"
         assert res.lei == "ZYXWVUTSRQP"
         assert res.institution_snapshot_id == "Snapshot-2"
+        assert res.creator.id == 4
+        # assert res.creator.user_id == "123456-7890-ABCDEF-GHIJ"
+        # assert res.creator.user_name == "test creator"
 
```

Comment:
remove the creator.id check and use these commented out ones, that's what we would want

Reply:
Oh I forgot about this one.